arch/risc-v/src/common/supervisor/riscv_sbi.c

```diff
@@ -63,23 +65,34 @@ static inline uintptr_t sbi_ecall(unsigned int extid, unsigned int fid,
      : "memory"
      );
 
-  return r1;
+  ret.error = r0;
+  ret.error = (uintreg_t)r1;
+
+  return ret;
 }
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
-/****************************************************************************
- * Name: riscv_sbi_set_timer
- *
- * Description:
- *   Set new compare match value for timer
- *
- * Input Parameters:
- *   stime_value - Value to set
- *
- ****************************************************************************/
+int sbi_err_map_errno(intreg_t err)
+{
+  switch (err)
+  {
+    case SBI_SUCCESS:
+      return 0;
+    case SBI_ERR_DENIED:
+      return -EPERM;
+    case SBI_ERR_INVALID_PARAM:
+      return -EINVAL;
+    case SBI_ERR_INVALID_ADDRESS:
+      return -EFAULT;
+    case SBI_ERR_NOT_SUPPORTED:
+    case SBI_ERR_FAILED:
+    default:
+      return -ENOTSUP;
```

Comment:
should we map all defined error code

Reply:
I think it is necessary. As the SBI error just starts for -1 and descends,  it is good to always have a default mapping for the new SBI error value to avoid confusing meaning.